src/explorer/Explorers/Components/AverageEstimator.cs

```diff
@@ -0,0 +1,56 @@
+namespace Explorer.Explorers.Components
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Diffix;
+    using Explorer.Common;
+    using Explorer.Explorers.Metrics;
+
+    internal class AverageEstimator : ExplorerComponent<AverageEstimator.Result>, DependsOn<NumericHistogramComponent.Result>
+    {
+        private ExplorerComponent<NumericHistogramComponent.Result>? histogramComponent;
+
+        public AverageEstimator(DConnection conn, ExplorerContext ctx)
+        : base(conn, ctx)
+        {
+        }
+
+        public void LinkToSourceComponent(ExplorerComponent<NumericHistogramComponent.Result> component)
+        {
+            histogramComponent = component;
+        }
+
+        protected override async Task<Result> Explore()
+        {
+            histogramComponent ??= new NumericHistogramComponent(Conn, Ctx);
```

Comment:
It shouldn't be necessary. The `??=` syntax means the component will be initialised only if it is `null`. Basically, if a component has already been linked, it will be used, otherwise a new one is initialised. 

So if you just do this at the top level without any linking:
```
Initialize<AverageEstimator, AverageEstimator.Result>();
```
Then it actually creates a HistogramComponent internally, and the HistogramComponent might also create its dependent component internally. However the resultant metrics of these 'internal' components are not published. 

Another approach would be to just throw an exception, which may be a better choice in some circumstances(?).

Reply:
Thanks for the clarifications, I missed the details on how the component created internally will not publish any metrics; this will be quite useful.